client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,228 @@
+import { Club } from '../../context/'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
+import { BasicEvent } from '../../context/Base'
+import moment from 'moment'
+
+interface BannerProps {
+  backgroundImageUrl: string
+}
+
+const Banner = styled.div`
+  background-image: url(${(props: BannerProps) => props.backgroundImageUrl || ''});
+  background-size: cover;
+  height: 400px;
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0 auto 100px;
+  width: 65%;
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
+
+  const { id } = router.query
+  if (typeof id !== 'string') return null
+
+  const club = clubs.get(parseInt(id))
+
+  return (
+    <div>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
+    </div>
+  )
+}
+
+const IconContainer = styled.div`
+  margin-top: 30px;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 20px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-top: 30px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 20px;
+`
+
+const SocialLink = styled.div`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 30px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const Categories = styled.div`
+  display: flex;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  font-size: 12px;
+  margin-left: 5px;
+
+  padding: 5px 10px;
+  @media ${device.mobileS} {
+    padding: 2px 5px;
+  }
+`
+
+const ClubDescription = styled.p`
+  margin-top: 30px;
+`
+
+interface ClubInfoProps {
+  club: Club
+}
+
+const ClubInfo = ({ club }: ClubInfoProps) => {
+  const { name, description, iconURL, tags } = club
+
+  return (
+    <div>
+      <IconContainer>
+        <FontAwesomeIcon icon={faLongArrowAltLeft} size="lg" />
+      </IconContainer>
+
+      <Logo src={iconURL} alt="" />
+      <ClubName>{name}</ClubName>
+
+      <Links>
+        <SocialLink>techplusuw.ca</SocialLink>
+        <SocialLink style={{ marginLeft: '10px' }}>@techplusuw</SocialLink>
+      </Links>
+
+      <ClubMetaData>
+        <MemberCount>20-30 members</MemberCount>
+
+        {/* TODO: color code the tags */}
+        <Categories>
+          {tags.map((t) => (
+            <Category>{t}</Category>
+          ))}
+        </Categories>
+      </ClubMetaData>
+
+      <ClubDescription>{description}</ClubDescription>
+    </div>
+  )
+}
+
+const EventsTitle = styled.h2``
```

Comment:
do we want to create empty styled components? super nitty, but I kinda see the value of both approaches (readability vs bloating codebase).

Reply:
I strongly prefer empty named styles. Adding customizations in the future is much easier if the style already exists

Reply:
Yeh that makes sense.